vlanBridge.go

```diff
@@ -436,3 +442,36 @@ func (vl *VlanBridge) processArp(pkt protocol.Ethernet, inPort uint32) {
 		}
 	}
 }
+
+// SendGARP sends GARP for the specified IP, MAC
+func (vl *VlanBridge) SendGARP(ip net.IP, mac net.HardwareAddr, vlanID uint16) error {
+	garpPkt, _ := protocol.NewARP(protocol.Type_Request)
+	garpPkt.HWSrc = mac
+	garpPkt.IPSrc = ip
+	garpPkt.HWDst, _ = net.ParseMAC("00:00:00:00:00:00")
+	garpPkt.IPDst = ip
+
+	log.Infof("Sending Gratuitous ARP request: %+v", garpPkt)
+
+	// Build the ethernet packet
+	ethPkt := protocol.NewEthernet()
+	ethPkt.VLANID.VID = vlanID
+	ethPkt.HWDst, _ = net.ParseMAC("FF:FF:FF:FF:FF:FF")
+	ethPkt.HWSrc = mac
+	ethPkt.Ethertype = 0x0806
+	ethPkt.Data = garpPkt
+
+	log.Debugf("Sending Gratuitous ARP request Ethernet: %+v", ethPkt)
+
+	// Construct Packet out
+	pktOut := openflow13.NewPacketOut()
+	pktOut.Data = ethPkt
+	for _, portNo := range vl.uplinkDb {
+		log.Debugf("Sending to uplink: %+v", portNo)
+		pktOut.AddAction(openflow13.NewActionOutput(portNo))
```

Comment:
@jojimt: We are trying to contain the ARP broadcasts to container and still send it to uplinks. From my discussion with @shaleman, the understanding is that in case of multiple links, we support only bonding interface type. In this case this should not be an issue.